pkg/daemon/ceph/osd/remove.go

```diff
@@ -138,5 +138,22 @@ func removeOSD(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInf
 		logger.Errorf("failed to remove CRUSH host %q. %v", hostName, err)
 	}
 
+	// The warning should be silenced by archiving the crash
+	crash, err := client.GetCrashList(clusterdContext, clusterInfo)
+	if err != nil {
+		logger.Errorf("failed to list ceph crash. %v", err)
```

Comment:
Need to `return` here instead of continuing with archiving the crash after the failure.

Reply:
Can please explain this in detail.

Reply:
If we fail to get the crash list, then we can't iterate over the crashes. 
```suggestion
		return errors.Wrap(err, "failed to list ceph crash")
```
Also if this code block is moved to a new method, returning from this failure won't skip any other operations for the osd removal. For example, we still want to print "completed removal of OSD ..." even if this crash removal operation failed.